### PR TITLE
#85: Add credential-gated live GitHub smoke tests

### LINKED ISSUE #85: Add credential-gated live GitHub smoke tests
## Issue Setup

- UAT Required: No
- Related Parent Issue or Context: Dogfood readiness gap for credential-gated live smoke tests

## Issue Goal

Add skipped-by-default credential-gated live GitHub smoke tests for Jade Symphony's Project v2 workflow.

## Why Now

Jade Symphony has live GitHub Project v2 reads/writes and a supervised dogfood smoke command, but the repo still lacks credential-gated integration tests that operators can run deliberately before dogfooding.

## Issue Context

Local `cargo test` must stay deterministic and credential-free. Live smoke coverage should only run when an explicit environment flag is set, and it must avoid mutating tracker state in this first slice.

### Decisions

- Use an explicit environment variable to opt into live smoke tests.
- Keep this first slice read-only / dry-run only.

### Assumptions

- `gh auth status` and the existing workflow's GitHub Project v2 config are enough to validate live read readiness.

## Non-Negotiable Guardrails

- Do not mutate GitHub Project state in these tests.
- Do not read or log secrets.
- Do not make default `cargo test` depend on network or credentials.
- Do not edit files under `docs/bootstrap/references/openai-symphony`.

### In Scope

- Add integration tests in a new integration-test directory that skip unless a live-smoke environment flag is set.
- Exercise `jade-symphony inspect examples/github-project-workflow.md`.
- Exercise `jade-symphony dogfood-smoke examples/github-project-workflow.md --dry-run`.
- Add a short docs page explaining how to run the credential-gated smoke tests.

### Out of Scope

- Live status mutation tests.
- Live workpad write tests.
- Linear credential-gated smoke tests.
- CI configuration changes.

### Target Repository / Package

- `Alive24/jade-symphony`

### Relevant Knowledge Sources

- `docs/dogfood-readiness.md`
- `docs/dogfood-smoke.md`
- `examples/github-project-workflow.md`

### Relevant Code Paths

- `src/main.rs`
- `src/tracker.rs`

## Current State

The repo has unit tests and fixture-backed dry-run coverage, but no credential-gated integration test harness for live GitHub Project v2 smoke checks.

## Deliverable Shape

Skipped-by-default integration tests plus a concise operator runbook.

## Risks or Constraints

- Live tests must be safe to run locally without altering tracker state.
- Test output must not include tokens or secrets.
- Missing credentials should produce a skipped test, not a failure, unless the opt-in flag is set.

## Expected Outcome

Operators can intentionally run live read/dry-run smoke checks with an environment flag while ordinary `cargo test` remains fast and credential-free.

### Completion Criteria

- Default `cargo test` passes with live tests skipped.
- Opt-in behavior is documented.
- The tests invoke the real binary instead of duplicating CLI logic.

### Functional Verification

- `cargo test`
- `cargo fmt --check`
- `cargo clippy --all-targets --all-features -- -D warnings`

### UAT

- Not required.

### Context Verification

- Confirm the tests are read-only / dry-run and do not require tracker writes.

### PR DESCRIPTION
**Pull request by Alive24**
## Summary

- Add skipped-by-default live GitHub smoke integration tests.
- Gate live execution behind `JADE_LIVE_GITHUB_SMOKE=1`.
- Exercise read-only `inspect` and dry-run `dogfood-smoke` against the live workflow.
- Add a short runbook for operators.

## Verification

- `cargo test`
- `cargo fmt --check`
- `cargo clippy --all-targets --all-features -- -D warnings`
- `JADE_LIVE_GITHUB_SMOKE=1 cargo test --test live_github_smoke`

Closes #85

### CODE REVIEW

**Review by Alive24 — COMMENTED** [review]
## 📋 Review Summary
Skipped due to error: Command '['gemini', '-p', '\nYou are a Code Review Architect. Follow these guidelines strictly:\n---\nname: code-review-commons\ndescription: Common guidelines, persona and critical constraints for performing high-quality code reviews. Use this skill when performing a /code-review or /pr-code-review command.\nuser-invokable: false\n---\n\n# Code Review Commons\n\n## PERSONA\n\nYou are a very experienced **Principal Software Engineer** and a meticulous **Code Review Architect**. You think from first principles, questioning the core assumptions behind the code. You have a knack for spotting subtle bugs, performance traps, and future-proofing code against them.\n\n## OBJECTIVE\n\nYour task is to deeply understand the **intent and context** of the provided code changes (diff content) and then perform a **thorough, actionable, and objective** review.\nYour primary goal is to **identify potential bugs, security vulnerabilities, performance bottlenecks, and clarity issues**.\nProvide **insightful feedback** and **concrete, ready-to-use code suggestions** to maintain high code quality and best practices. Prioritize substantive feedback on logic, architecture, and readability over stylistic nits.\n\n## Instructions\n\n1. **Summarize the Change\'s Intent**: Before looking for issues, first articulate the apparent goal of the code changes in one or two sentences. Use this understanding to frame your review.\n2. **Establish context** by reading relevant files. Prioritize:\n    a. All files present in the diff.\n    b. Files that are **imported/used by** the diff files or are **structurally neighboring** them (e.g., related configuration or test files).\n3. **Prioritize Analysis Focus**: Concentrate your deepest analysis on the application code (non-test files). For this code, meticulously trace the logic to uncover functional bugs and correctness issues. Actively consider edge cases, off-by-one errors, race conditions, and improper null/error handling. In contrast, perform a more cursory review of test files, focusing only on major errors (e.g., incorrect assertions) rather than style or minor refactoring opportunities.\n4. **Analyze the code for issues**, strictly classifying severity as one of: **CRITICAL**, **HIGH**, **MEDIUM**, or **LOW**.\n\n## Critical Constraints\n\n**STRICTLY follow these rules for review comments:**\n\n* **Location:** You **MUST** only provide comments on lines that represent actual changes in the diff. This means your comments must refer **only to lines beginning with `+` or `-`**. **DO NOT** comment on context lines (lines starting with a space).\n* **Relevance:** You **MUST** only add a review comment if there is a demonstrable **BUG**, **ISSUE**, or a significant **OPPORTUNITY FOR IMPROVEMENT** in the code changes.\n* **Tone/Content:** **DO NOT** add comments that:\n    * Tell the user to "check," "confirm," "verify," or "ensure" something.\n    * Explain what the code change does or validate its purpose.\n    * Explain the code to the author (they are assumed to know their own code).\n    * Comment on missing trailing newlines or other purely stylistic issues that do not affect code execution or readability in a meaningful way.\n* **Substance First:** **ALWAYS** prioritize your analysis on the **correctness** of the logic, the **efficiency** of the implementation, and the **long-term maintainability** of the code.\n* **Technical Detail:**\n    * Pay **meticulous attention to line numbers and indentation** in code suggestions; they **must** be correct and match the surrounding code.\n    * **NEVER** comment on license headers, copyright headers, or anything related to future dates/versions (e.g., "this date is in the future").\n* **Formatting/Structure:**\n    * Keep the **change summary** concise (aim for a single sentence).\n    * Keep **comment bodies concise** and focused on a single issue.\n    * If a similar issue exists in **multiple locations**, state it once and indicate the other locations instead of repeating the full comment.\n    * **AVOID** mentioning your instructions, settings, or criteria in the final output.\n\n**Severity Guidelines (for consistent classification):**\n\n* **Functional correctness bugs that lead to behavior contrary to the change\'s intent should generally be classified as HIGH or CRITICAL.**\n* **CRITICAL:** Security vulnerabilities, system-breaking bugs, complete logic failure.\n* **HIGH:** Performance bottlenecks (e.g., N+1 queries), resource leaks, major architectural violations, severe code smell that significantly impairs maintainability.\n* **MEDIUM:** Typographical errors in code (not comments), missing input validation, complex logic that could be simplified, non-compliant style guide issues (e.g., wrong naming convention).\n* **LOW:** Refactoring hardcoded values to constants, minor log message enhancements, comments on docstring/Javadoc expansion, typos in documentation (.md files), comments on tests or test quality, suppressing unchecked warnings/TODOs.\n\n\nReview this PR.\nTitle: #85: Add credential-gated live GitHub smoke tests\nBody: ## Summary\n\n- Add skipped-by-default live GitHub smoke integration tests.\n- Gate live execution behind `JADE_LIVE_GITHUB_SMOKE=1`.\n- Exercise read-only `inspect` and dry-run `dogfood-smoke` against the live workflow.\n- Add a short runbook for operators.\n\n## Verification\n\n- `cargo test`\n- `cargo fmt --check`\n- `cargo clippy --all-targets --all-features -- -D warnings`\n- `JADE_LIVE_GITHUB_SMOKE=1 cargo test --test live_github_smoke`\n\nCloses #85\n\n\nDiff:\n```diff\ndiff --git a/docs/dogfood-readiness.md b/docs/dogfood-readiness.md\nindex 305d110..f2bbc35 100644\n--- a/docs/dogfood-readiness.md\n+++ b/docs/dogfood-readiness.md\n@@ -27,7 +27,7 @@ yet.\n | Project doctor | Read-only `doctor` / `audit-project` reports workflow invariant violations from normalized tracker issues, including missing PR handoff evidence, missing review pass evidence, dirty Merging PRs, missing runtime ownership hints, and queued issues with PRs. Repair mode is not implemented yet. |\n | Observability | Operator-readable terminal snapshots report polling, running, retrying, skipped issues, gate details, token counters, event-log path, and integration gaps. JSONL event-log primitives exist and `run-once` writes dry-run events with actor and profile metadata. Runtime state files are written during write-mode `run-loop` issue execution, including actor role/label, git author, and optional profile/instance identity when configured; resume, retry, usage-limit pause, and stall supervision events are also recorded. No web/API surface yet. |\n | Usage-limit pause/resume | Conservative usage-limit/rate-limit/resource-exhausted classification exists for subprocess agent events and review job output. `run-loop` records usage-limit pauses in workpad/runtime retry state and does not advance to `Agent Review`; review workpads surface usage-limit failures without moving to `Human Review`. Vendor-specific quota management is not implemented. |\n-| Tests | Unit tests cover the dry-run skeleton. No credential-gated integration tests yet. |\n+| Tests | Unit tests cover the dry-run skeleton. Read-only / dry-run live GitHub smoke tests exist behind explicit `JADE_LIVE_GITHUB_SMOKE=1` opt-in; mutation and Linear credential-gated smoke coverage are still missing. |\n \n The operator launcher runbook is in `docs/operator-dogfood.md`; it keeps write\n mode explicit through `scripts/jade-dogfood --write --confirm-write`.\ndiff --git a/docs/live-github-smoke-tests.md b/docs/live-github-smoke-tests.md\nnew file mode 100644\nindex 0000000..d574019\n--- /dev/null\n+++ b/docs/live-github-smoke-tests.md\n@@ -0,0 +1,38 @@\n+# Live GitHub Smoke Tests\n+\n+These integration tests are opt-in checks for a real GitHub Project v2\n+workflow. They are read-only / dry-run only in this slice, and ordinary\n+`cargo test` skips them without credentials.\n+\n+## Prerequisites\n+\n+- `gh` is installed.\n+- `gh auth status` succeeds for the `Alive24/jade-symphony` repository.\n+- The workflow at `examples/github-project-workflow.md` points at the intended\n+  GitHub Project v2 tracker.\n+\n+No tokens or secrets are printed by the tests.\n+\n+## Run\n+\n+```bash\n+JADE_LIVE_GITHUB_SMOKE=1 cargo test --test live_github_smoke\n+```\n+\n+The smoke runs:\n+\n+- `jade-symphony inspect examples/github-project-workflow.md`\n+- `jade-symphony dogfood-smoke examples/github-project-workflow.md --dry-run`\n+\n+## Expected Behavior\n+\n+- Without `JADE_LIVE_GITHUB_SMOKE=1`, the tests print a skip message and pass.\n+- With the flag set, missing or unusable `gh` authentication is a test failure.\n+- The smoke must not mutate GitHub Project state, workpad comments, issues, or\n+  PRs.\n+\n+## Follow-Ups\n+\n+Mutation smoke tests for status updates, workpad upsert, and controlled\n+worktree/PR handoff should remain separate opt-in slices with their own\n+guardrails.\ndiff --git a/tests/live_github_smoke.rs b/tests/live_github_smoke.rs\nnew file mode 100644\nindex 0000000..04f900a\n--- /dev/null\n+++ b/tests/live_github_smoke.rs\n@@ -0,0 +1,82 @@\n+use std::process::{Command, Output};\n+\n+const LIVE_SMOKE_ENV: &str = "JADE_LIVE_GITHUB_SMOKE";\n+const WORKFLOW: &str = "examples/github-project-workflow.md";\n+\n+fn live_smoke_enabled() -> bool {\n+    matches!(\n+        std::env::var(LIVE_SMOKE_ENV).as_deref(),\n+        Ok("1") | Ok("true") | Ok("yes")\n+    )\n+}\n+\n+fn repo_root() -> &\'static str {\n+    env!("CARGO_MANIFEST_DIR")\n+}\n+\n+fn combined_output(output: &Output) -> String {\n+    format!(\n+        "stdout:\\n{}\\nstderr:\\n{}",\n+        String::from_utf8_lossy(&output.stdout),\n+        String::from_utf8_lossy(&output.stderr)\n+    )\n+}\n+\n+fn require_live_smoke() -> bool {\n+    if !live_smoke_enabled() {\n+        eprintln!("skipping live GitHub smoke; set {LIVE_SMOKE_ENV}=1 to enable");\n+        return false;\n+    }\n+\n+    let output = Command::new("gh")\n+        .args(["auth", "status"])\n+        .current_dir(repo_root())\n+        .output()\n+        .expect("failed to execute `gh auth status`");\n+    assert!(\n+        output.status.success(),\n+        "live smoke requires usable gh auth\\n{}",\n+        combined_output(&output)\n+    );\n+    true\n+}\n+\n+fn run_jade(args: &[&str]) -> Output {\n+    Command::new(env!("CARGO_BIN_EXE_jade-symphony"))\n+        .args(args)\n+        .current_dir(repo_root())\n+        .output()\n+        .expect("failed to execute jade-symphony binary")\n+}\n+\n+#[test]\n+fn live_github_project_inspect_smoke() {\n+    if !require_live_smoke() {\n+        return;\n+    }\n+\n+    let output = run_jade(&["inspect", WORKFLOW]);\n+    let text = combined_output(&output);\n+    assert!(output.status.success(), "{text}");\n+    assert!(text.contains("issues="), "{text}");\n+    assert!(\n+        !text.contains("no usable GitHub auth was detected"),\n+        "inspect should not report missing auth when live smoke is enabled\\n{text}"\n+    );\n+}\n+\n+#[test]\n+fn live_github_dogfood_smoke_dry_run_is_read_only() {\n+    if !require_live_smoke() {\n+        return;\n+    }\n+\n+    let output = run_jade(&["dogfood-smoke", WORKFLOW, "--dry-run"]);\n+    let text = combined_output(&output);\n+    assert!(output.status.success(), "{text}");\n+    assert!(text.contains("dogfood_smoke="), "{text}");\n+    assert!(\n+        text.contains("dry_run") || text.contains("blocked"),\n+        "dogfood smoke should remain dry-run/read-only in this smoke\\n{text}"\n+    );\n+}\n\n```\n\nWrite a review markdown file exactly in this format:\n## 📋 Review Summary\n[2-3 sentences summarizing the PR]\n## 🔍 General Feedback\n- [Feedback point 1]\n- [Feedback point 2]\n<COMMENT>\n[SEVERITY] [COMMENT_TEXT]\n</COMMENT>\n\n(Only add <COMMENT> if there\'s inline feedback). Do not include any other text before or after.\n']' timed out after 60 seconds
## 🔍 General Feedback
- None

**Review by Alive24 — COMMENTED** [review]
## 📋 Review Summary

This PR adds valuable credential-gated live smoke tests for GitHub integrations, providing high-confidence verification of the tracker adapter.

## 🔍 General Feedback

- **Live Verification**: Adding these tests guarantees that the GitHub Project v2 adapter works against real API responses, catching schema drifts early.
- **Safe Execution**: Appropriately gating these behind credentials ensures they don't break standard developer or offline CI environments.
